xscen/ensembles.py

```diff
@@ -441,7 +452,72 @@ def generate_weights(
         # Drop the experiment coordinate
         weights = weights.drop_vars("experiment")
 
+    # Attribute_weights
+    if attribute_weights:
```

Comment:
With where this is placed and how it is implemented, this could create really unintuitive results if:

- Weights are applied along `experiment`, but `experiment_weights` is False.
  - e.g. If I have 10x SSP245 and 2x SSP585 with `experiment_weights=False`, then use `attribute_weights={experiment: {SSP245:0.5, SSP585:1}`, the total weight of SSP245 is still more than twice that of SSP585.
- There's a mismatch between the attribute used here and the independence level.
  - e.g. If I have GFDL-ESM2M, GFDL-ESM2G, and CSIRO with `independence_level=institution`, then use `attribute_weights={model: {GFDL-ESM2M:0.5, others:1}`, then I don't really have 1 institution 1 vote anymore.